fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/directory/AgilityContextTest.java

```diff
@@ -175,13 +202,14 @@ void testAgilityContextOneLongWrite(int loopCount, int sizeLimit, int timeLimit,
     @ParameterizedTest
     @BooleanSource
     void testAgilityContextSizeLimit(boolean useProp) {
-        testAgilityContextOneLongWrite(73, 100, 100000, useProp);
+        testAgilityContextOneLongWrite(73, 100, 100000, useProp, true);
     }
 
     @ParameterizedTest
     @BooleanSource
     void testAgilityContextTimeLimit(boolean useProp) {
-        testAgilityContextOneLongWrite(77, 100000, 10, useProp);
+        // The 1ms time limit often triggers an auto-commit, that that cannot be guaranteed here.
+        testAgilityContextOneLongWrite(77, 100000, 1, useProp, false);
```

Comment:
How close are we to 1ms? Even though it could be slightly, I feel like having a 2ms sleep in the loop, and asserting that it did auto-commit would be better than not testing that we check, and apply the time limit at all.

Reply:
✅